FrontEnd/src/UserAuthentication/SignUpPage.jsx

```diff
@@ -1,77 +1,145 @@
-import  {useNavigate} from "react-router-dom";
-import {useState} from "react";
-import {registerUser} from "./Auth";
-import {addUser} from './FirestoreDB';
-import './SignUpPage.css';
+import { useNavigate, Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { registerUser } from "./Auth";
+import "./SignUpPage.css";
 
-function SignUpPage(){
-    const navigate = useNavigate();
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
-    const [firstName, setFirstName] = useState('');
-    const [lastName, setLastName] = useState('');
-    const [userName, setUserName] = useState('');
-    const MIN_PASSWORD_LENGTH = 6;
-    const handleLoginRoute = () => {
-        navigate(`/LoginPage`);
-      };
+function SignUpPage() {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [userName, setUserName] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [isFormValid, setIsFormValid] = useState(false);
+  const MIN_PASSWORD_LENGTH = 8;
 
-    const handleSignUp = async(event) =>{
-        event.preventDefault();
-        if (password.length < MIN_PASSWORD_LENGTH) {
-            alert(`Minimum password length is ${MIN_PASSWORD_LENGTH}`);
-        } else {
-            try{
-                const userCredential = await registerUser(email, password);
-                if (userCredential!==undefined){
-                    addUser(userCredential, firstName, lastName, userName, email);
-                    navigate('/MainPage');
-                } else{
-                    alert("Incorrect Email address");
-                };
-            } catch (error){
-                console.error("Error registering user:", error);
-            };
-        };
-    };
-    return(
+  const validateForm = () => {
+    return (
+      email.trim() !== "" && password.trim() !== "" && userName.trim() !== ""
+    );
+  };
+  useEffect(() => {
+    setIsFormValid(validateForm());
+  }, [email, password, userName]);
+
+  const handleSignUp = async (event) => {
+    event.preventDefault();
+    if (password.length < MIN_PASSWORD_LENGTH) {
+      setErrorMessage("Password must be at least 8 characters");
```

Comment:
You can use `MIN_PASSWORD_LENGTH` in the error message, which is useful if you later on change how many characters are needed for the password.

`setErrorMessage(`Password must be at least ${MIN_PASSWORD_LENGTH} characters`);`